packages/serverpod/lib/src/server/websocket_request_handlers/endpoint_websocket_request_handler.dart

```diff
@@ -141,34 +141,38 @@ abstract class EndpointWebsocketRequestHandler {
 
   static Future<void> _callStreamOpened(
     StreamingSession session,
-    Endpoint endpoint,
+    String endpointName,
+    EndpointDispatch endpointDispatch,
   ) async {
     try {
-      session.endpoint = endpoint.name;
-      var authFailed = await EndpointDispatch.canUserAccessEndpoint(
-        () => session.authenticated,
-        endpoint.requireLogin,
-        endpoint.requiredScopes,
+      session.endpoint = endpointName;
+      var connector = await endpointDispatch.getEndpointConnector(
+        session: session,
+        endpointPath: endpointName,
       );
-      if (authFailed == null) await endpoint.streamOpened(session);
-    } catch (e) {
+      await connector.endpoint.streamOpened(session);
+    } on NotAuthorizedException {
+      return;
+    } catch (_) {
       return;
     }
   }
 
   static Future<void> _callStreamClosed(
     StreamingSession session,
-    Endpoint endpoint,
+    String endpointName,
+    EndpointDispatch endpointDispatch,
   ) async {
     try {
-      session.endpoint = endpoint.name;
-      var authFailed = await EndpointDispatch.canUserAccessEndpoint(
-        () => session.authenticated,
-        endpoint.requireLogin,
-        endpoint.requiredScopes,
+      session.endpoint = endpointName;
+      var connector = await endpointDispatch.getEndpointConnector(
+        session: session,
+        endpointPath: endpointName,
       );
-      if (authFailed == null) await endpoint.streamClosed(session);
-    } catch (e) {
+      await connector.endpoint.streamClosed(session);
+    } on NotAuthorizedException {
+      return;
+    } catch (_) {
       return;
     }
```

Comment:
Comment: just noticed that it is extremely important that we catch any exception here, otherwise we won't close the session since this happens after all calls to `_callStreamClosed`.